android/build.gradle

```diff
@@ -32,18 +79,24 @@ android {
 }
 
 repositories {
-    if (project == rootProject) {
-        maven {
-            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
-            url "$rootDir/../node_modules/react-native/android"
-        }
-        google()
-        jcenter()
+    maven {
+        // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
+        url "$rootDir/../node_modules/react-native/android"
     }
+    mavenCentral()
+    google()
 }
 
 dependencies {
     //noinspection GradleDynamicVersion
     implementation "com.facebook.react:react-native:+"
     implementation 'androidx.appcompat:appcompat:1.2.0'
 }
+
+if (isNewArchitectureEnabled()) {
+    react {
```

Comment:
For monorepo setups I believe you need to pass reactNativeDir/codegenDir too as in the linked example